packages/shorebird_cli/lib/src/commands/release_new/android_releaser.dart

```diff
@@ -0,0 +1,212 @@
+import 'package:mason_logger/mason_logger.dart';
+import 'package:shorebird_cli/src/artifact_builder.dart';
+import 'package:shorebird_cli/src/code_push_client_wrapper.dart';
+import 'package:shorebird_cli/src/commands/release_new/release_type.dart';
+import 'package:shorebird_cli/src/commands/release_new/releaser.dart';
+import 'package:shorebird_cli/src/doctor.dart';
+import 'package:shorebird_cli/src/logger.dart';
+import 'package:shorebird_cli/src/platform.dart';
+import 'package:shorebird_cli/src/platform/platform.dart';
+import 'package:shorebird_cli/src/shorebird_android_artifacts.dart';
+import 'package:shorebird_cli/src/shorebird_flutter.dart';
+import 'package:shorebird_cli/src/shorebird_validator.dart';
+import 'package:shorebird_cli/src/third_party/flutter_tools/lib/flutter_tools.dart';
+import 'package:shorebird_cli/src/third_party/flutter_tools/lib/src/base/io.dart';
+import 'package:shorebird_cli/src/version.dart';
+import 'package:shorebird_code_push_client/shorebird_code_push_client.dart';
+
+/// {@template android_releaser}
+/// Functions to create an Android release.
+/// {@endtemplate}
+class AndroidReleaser extends Releaser {
+  /// {@macro android_releaser}
+  AndroidReleaser({
+    required super.argResults,
+    required super.flavor,
+    required super.target,
+  });
+
+  @override
+  ReleaseType get releaseType => ReleaseType.android;
+
+  /// The architectures to build for.
+  Set<Arch> get architectures => (argResults['target-platform'] as List<String>)
+      .map(
+        (platform) => AndroidArch.availableAndroidArchs
+            .firstWhere((arch) => arch.targetPlatformCliArg == platform),
+      )
+      .toSet();
+
+  /// Whether to generate an APK in addition to the AAB.
+  late bool generateApk = argResults['android-artifact'] as String == 'apk';
```

Comment:
Should we have an `generateAar` as well?

Reply:
Nope – aar is a completely different pathway (it's the equivalent of `flutter build aar` vs `flutter build appbundle`)